lib/web/desktop.go

```diff
@@ -84,12 +122,22 @@ func (h *Handler) handleDesktopAccessWebsocket(
 	serviceConTLS := tls.Client(serviceCon, ctx.clt.Config())
 	log.Debug("Connected to windows_desktop_service")
 
+	tdpConn := tdp.NewConn(serviceConTLS)
+	err = tdpConn.OutputMessage(tdp.ClientUsername{Username: username})
```

Comment:
Do these messages have to happen in `proxyWebsocketConn` after we set `ws.PayloadType` but before we start proxying data?

Reply:
I don't quite see it... Why might they? AFAIK the `ws.PayloadType` isn't relevant at this point, I'm just writing directly to (4)

![Image from iOS](https://user-images.githubusercontent.com/13578537/144107993-e6718a46-2e1f-4430-bbcc-07a84494cb3c.jpg)

I do wonder, though, if there may be a race condition here: [once I write those two messages](https://github.com/gravitational/teleport/blob/master/lib/srv/desktop/rdp/rdpclient/client.go#L145-L153), the RDP client [connects](https://github.com/gravitational/teleport/blob/d5b3cdd8ebde274fd022015e06538971b1d012b9/lib/srv/desktop/rdp/rdpclient/client.go#L196) and then [begins sending pngs](https://github.com/gravitational/teleport/blob/d5b3cdd8ebde274fd022015e06538971b1d012b9/lib/srv/desktop/rdp/rdpclient/client.go#L231). 

If that process all happens very fast before `proxyWebsocketConn` has run, is there a chance that some of the initial png's are never read and sent over the websocket? The read buffer of the `serviceConTLS` must have some limit...

Reply:
Ah, you're right. Thanks for using my own diagram to teach me :) 

I don't see a race, should be fine.